proxy/envoy/discovery.go

```diff
@@ -302,6 +321,37 @@ func (ds *DiscoveryService) clearCache() {
 	ds.rdsCache.clear()
 }
 
+// ListAllEndpoints responds with all Services and is restricted to a single service key
+func (ds *DiscoveryService) ListAllEndpoints(request *restful.Request, response *restful.Response) {
+
+	var servicesArray []*service
+	for _, svc := range ds.services.Services() {
+		for _, port := range svc.Ports {
+
+			var hostArray []*host
+			for _, ep := range ds.services.Instances(svc.Hostname, []string{port.Name}, []model.Tags{}) {
```

Comment:
Since we know the size out the output array, you can pre-allocate the space that you'll need.
```
instances := ds.services.Instances(...)
hosts := make([]*host, 0, len(hosts))
for _, ep := range instances {
   ...
}
```